FHEM/38_co2mini.pm

```diff
@@ -80,27 +69,24 @@ co2mini_Connect($)
   return undef if( AttrVal($name, "disable", 0 ) == 1 );
 
   # Reasonably low certainty that something:onlynumbers is a device node, but could be network address
-  if($hash->{DEVICE} =~ /^(.*):(\d+)$/) {
-    $hash->{HANDLE} = IO::Socket::INET->new(PeerAddr=>$hash->{DEVICE}, Proto=>"tcp", Blocking=>0) or return "Error opening " . $hash->{DEVICE};
+  if($hash->{DeviceName} =~ /^(.*):(\d+)$/) {
     $hash->{helper}{mode} = "net";
     $hash->{helper}{buf} = "";
-  } else {
-    sysopen($hash->{HANDLE}, $hash->{DEVICE}, O_RDWR | O_APPEND | O_NONBLOCK) or return "Error opening " . $hash->{DEVICE};
+    return DevIo_OpenDev($hash, 1, "co2mini_OnConnect");
+  } elsif($hash->{helper}{mode} eq "dev") {
```

Comment:
Ich habe da gerade nochmal drauf geguckt und ich sehe da gute Chancen, mit der Syntax "/dev/co2mini0@directio". Er nutzt dann Perl's open auf den Devicenamen. Das Handle ist dann per $hash->{DIODev} verfügbar. Ich bin mir aber sicher, dass du das auch schon rausgefunden hast...
Sorry, dass ich da nicht selbst mitspielen kann. VMWare weigert sich leider standhaft den CO2mini in meine VM zu connecten :(